Kite SDK/PSPrintSDK/OLKiteViewController.m

```diff
@@ -132,61 +149,76 @@ -(IBAction) dismiss{
-    [vc safePerformSelector:@selector(setUserSelectedPhotos:) withObject:self.userSelectedPhotos];
-    [vc safePerformSelector:@selector(setTemplateClass:) withObject:product.productTemplate.templateClass];
-    [[vc navigationItem] setLeftBarButtonItem:[[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:self action:@selector(dismiss)]];
-    
-    [self fadeToViewController:nav];
+        else if (welf.printOrder && welf.showProductDescriptionWithPrintOrder){
+            OLProductOverviewViewController *vc = [welf.storyboard instantiateViewControllerWithIdentifier:@"OLProductOverviewViewController"];
+            vc.product = [OLProduct productWithTemplateId:[[welf.printOrder.jobs firstObject] templateId]];
+            vc.userEmail = welf.userEmail;
+            vc.userPhone = welf.userPhone;
+            vc.delegate = welf.delegate;
+            [[vc navigationItem] setLeftBarButtonItem:[[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:welf action:@selector(dismiss)]];
+            [vc.navigationItem.rightBarButtonItem setTitle:NSLocalizedString(@"Next", @"")];
+            OLCustomNavigationController *nvc = [[OLCustomNavigationController alloc] initWithRootViewController:vc];
+            [welf fadeToViewController:nvc];
+            return;
+        }
+        else if (groups.count == 1) {
+            OLProductGroup *group = groups[0];
+            product = [group.products firstObject];
+            nextVcNavIdentifier = [OLKiteViewController storyboardIdentifierForGroupSelected:group];
+        }
+        else {
+            // Launch the product home view controller where the top level groups will be displayed
+            nextVcNavIdentifier = @"ProductHomeViewController";
+        }
+        UIViewController *vc = [sb instantiateViewControllerWithIdentifier:nextVcNavIdentifier];
+        UINavigationController *nav = [[OLCustomNavigationController alloc] initWithRootViewController:vc];
+        [vc safePerformSelector:@selector(setProduct:) withObject:product];
+        [vc safePerformSelector:@selector(setDelegate:) withObject:welf.delegate];
+        [vc safePerformSelector:@selector(setUserEmail:) withObject:welf.userEmail];
+        [vc safePerformSelector:@selector(setUserPhone:) withObject:welf.userPhone];
+        [vc safePerformSelector:@selector(setFilterProducts:) withObject:welf.filterProducts];
+        [vc safePerformSelector:@selector(setUserSelectedPhotos:) withObject:welf.userSelectedPhotos];
+        [vc safePerformSelector:@selector(setTemplateClass:) withObject:product.productTemplate.templateClass];
+        [[vc navigationItem] setLeftBarButtonItem:[[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel target:welf action:@selector(dismiss)]];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [welf fadeToViewController:nav];
```

Comment:
This is how you should be doing it above. My suggestion though is to wrap the entire block in a dispatch async to keep it really clean and obvious. Alternatively I suggest for this you use [NSOperationQueue mainQueue] as it will do the right thing!